metrics/web/test/dashboard/domain/entities/metrics/receive_build_day_project_metrics_updates_test.dart

```diff
@@ -0,0 +1,263 @@
+            buildDay.failed +
+            buildDay.unknown +
+            buildDay.inProgress;
+      });
+    }
+
+    Duration calculateTotalBuildsDuration(List<BuildDay> buildDays) {
+      return buildDays.fold<Duration>(Duration.zero, (value, buildDay) {
+        return value + buildDay.totalDuration;
+      });
+    }
+
+    tearDown(() {
+      reset(buildDayRepository);
+    });
+
+    test(
+      "throws an ArgumentError if the given build day repository is null",
+      () {
+        expect(
+          () => ReceiveBuildDayProjectMetricsUpdates(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "successfully creates an instance with the given build day repository",
+      () {
+        expect(
+          () => ReceiveBuildDayProjectMetricsUpdates(buildDayRepository),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      "successfully creates an instance with the given build day repository",
+      () {
+        expect(
+          () => ReceiveBuildDayProjectMetricsUpdates(buildDayRepository),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      ".call() delegates to the .projectBuildDaysInDateRangeStream() with the given project id and a non-null from date",
+      () {
+        whenFetchBuildDaysStream().thenAnswer((_) => Stream.value([]));
+
+        useCase.call(projectIdParam);
+
+        verify(buildDayRepository.projectBuildDaysInDateRangeStream(
+          projectId,
+          from: argThat(isNotNull, named: 'from'),
+        )).called(once);
+      },
+    );
+
+    test(
+      ".call() returns a build day project metrics with the given project id",
+      () async {
+        whenFetchBuildDaysStream().thenAnswer((_) => Stream.value([]));
+
+        final metricsStream = useCase.call(projectIdParam);
+        final buildDayMetrics = await metricsStream.firstWhere(
+          (metrics) => metrics != null,
+        );
+
+        expect(buildDayMetrics.projectId, equals(projectId));
+      },
+    );
+
+    test(
+      ".call() returns a build day project metrics with the non-null build number metric",
+      () async {
+        whenFetchBuildDaysStream().thenAnswer((_) => Stream.value([]));
+
+        final metricsStream = useCase.call(projectIdParam);
+        final buildDayMetrics = await metricsStream.firstWhere(
+          (metrics) => metrics != null,
+        );
+
+        expect(buildDayMetrics.buildNumberMetric, isNotNull);
+      },
+    );
+
+    test(
+      ".call() returns a build day project metrics with the build number metric equal to the total number of builds performed during the returned build days",
+      () async {
+        final firstBuildDay = createBuildDay(successful: 1, failed: 1);
+        final secondBuildDay = createBuildDay(unknown: 2, inProgress: 2);
+        final buildDays = [firstBuildDay, secondBuildDay];
+        final expectedNumberOfBuilds = calculateTotalNumberOfBuilds(buildDays);
```

Comment:
The `calculateTotalNumberOfBuilds` function here is a potential item for testing by itself. Moreover, it is implemented in the same way as builds are calculated within the use case under tests. This makes tests confusing as they know how to calculate everything that is calculated inside the use case. Thus, we should either move a logic with the required calculation elsewhere and test it separately or use concrete values in tests. Here, we definitely know that the number of builds for the specified list is `1 + 1 + 2 + 2 = 6`. As we use **concrete** test data, we expect **concrete** results and we know them. The same idea is applied to the `calculateTotalBuildsDuration `.
```suggestion
        const expectedNumberOfBuilds = 1 + 1 + 2 + 2;
```
or, which is the same:
```suggestion
        const expectedNumberOfBuilds = 6;
```